pkg/execution/executor/executor.go

```diff
@@ -660,12 +694,25 @@ func (e *executor) Schedule(ctx context.Context, req execution.ScheduleRequest)
 		Payload: queue.PayloadEdge{
 			Edge: inngest.SourceEdge,
 		},
-		Throttle: throttle,
+		Throttle:  throttle,
+		Singleton: singletonConfig,
 	}
 	err = e.queue.Enqueue(ctx, item, at, queue.EnqueueOpts{})
 	if err == redis_state.ErrQueueItemExists {
 		return nil, state.ErrIdentifierExists
 	}
+	if err == redis_state.ErrQueueItemSingletonExists {
+		_, delErr := e.smv2.Delete(ctx, sv2.IDFromV1(st.Identifier()))
+
+		if delErr != nil {
+			logger.StdlibLogger(ctx).Error(
+				"error deleting function state",
+				"error", err,
+			)
+		}
+
+		return e.handleFunctionSkipped(ctx, req, metadata, evts, enums.SkipReasonSingleton)
+	}
```

Comment:
it's accumulating a bunch of `if` statements here now.
let's swap it out to a `switch` instead.